### PR TITLE
check error when the "image" field of json file of dataset is empty, which may lead to an empty dataloader

### PR DESCRIPTION
**Pull request by Xinyi-Adv**
#7105

### CODE REVIEW

File: mmdet/core/evaluation/eval_hooks.py [hunk 1]
```diff
@@ -53,6 +53,8 @@ def _do_evaluate(self, runner):
             return
 
         from mmdet.apis import single_gpu_test
+        # assert when dataloader is empty, in case that Error Occurs when running train.py with the "image" field of json file (val2017.json) set empty
```
Comment (RangiLyu) [inline mmdet/core/evaluation/eval_hooks.py:56]: Not all task uses json as an annotation file. Try to make the error message and comments be more specific.

File: mmdet/core/evaluation/eval_hooks.py [hunk 2]
```diff
@@ -53,6 +53,8 @@ def _do_evaluate(self, runner):
...
+        assert len(self.dataloader) != 0, "empty dataloader, which may be due to the \"image\" field of json file"
```
Comment (RangiLyu) [inline mmdet/core/evaluation/eval_hooks.py:57]: Adding the assertion directly after building dataloader instead of here would be better.

Comment (Xinyi-Adv) [reply mmdet/core/evaluation/eval_hooks.py:57]: ohhh Thank you. I have posted a new PR refering to your advice #7129

**Comment by hhaAndroid** [comment]
Duplicate. I will close it and open it myself if necessary

**Pull request closed without merging**